### ISSUE TITLE
Allow user to specify origin location on Globus

### ISSUE THREAD

**Issue by amyehodge** [sul-dlss/happy-heron#3022]
1. Allow the user to tell us if their content is on Google Drive, Oak, or Sherlock so that we can send a more precise link that will open the Globus file manager with the origin open on one side and the destination open on the other side. This will make getting to the right place in Globus much easier for users, because it's very challenging to navigate around the Globus interface.
2. Allow the user to tell us if their content is on a different endpoint (by not selecting any of the fixed options) ~~and provide us with the info we need to generate the more precise link to that endpoint. What we need from them may be the UUID for the endpoint, but we need to look into that.~~ 
3. The link in box/step 3 should go to https://docs.google.com/document/d/10b7y3yZCOfyVJ_uP4l7QHbkILQKnfdJi9hj-MpmmdIk/edit (top of the page) and not to a specific header (I'm not sure where that points now). Also, we are not including the "Print instructions" link show in this image.
Updated design: <img width="1073" alt="image" src="https://github.com/sul-dlss/happy-heron/assets/42216982/1692a902-a492-41ab-a34b-4597acb68172">

4. Make additional changes to the "steps" on top of deposit form as shown here:
https://projects.invisionapp.com/share/YQ132HXFX2GR#/screens/472346509_Globus_Auto_Account

See also #3116, #3021 and #3049 
See designs at https://projects.invisionapp.com/share/YQ132HXFX2GR#/screens/472346509_Globus_Auto_Account

Information from the folks at Globus on how to create these links:
 
- File Manager application links with pre-selected endpoints
— Query parameters supported: destination_id, destination_path, origin_id and origin_path
-- Example https://app.globus.org/file-manager?destination_id=ddb59aef-6d04-11e5-ba46-22000b92c6ec&destination_path=%2F~%2F&origin_id=2fde89c0-6fb4-11eb-8c47-0eb1aa8d4337

Note that we currently have Globus send the link to the shared folder, I think because we didn't know how to figure out what the direct link would be, but they say this is knowable:

origin_id is the endpoint id of source, origin_path is the source path. Those must be available from when you set permission for the user on a specific folder. You can use that to construct the URL.

**Comment by amyehodge** [comment]
@amyehodge needs to inquire with Karl Kornel what existing options there are, or how someone could provide us with a link for their GCP endpoint possibly.

@astridu Let's talk about the designs needed to implement this. For a list of things like Oak, Sherlock, etc. we may only need check boxes or a dropdown list. For providing a link to your own endpoint, that would require something else.

**Comment by peetucket** [comment]
@amyehodge 

1. Are items 2 and 3 in the description above the same?
2. Do we know what entering "other" does to the link we create?
3. Can I split these into two (or three) tickets?

**Comment by amyehodge** [comment]
@peetucket 

> * Are items 2 and 3 in the description above the same?

They are mocked up as part of the same option, but I consider it a higher priority to provide the three checkboxes for Sherlock, Oak, and Google Drive that for users to be able to give us a link to a specific other endpoint. So I think these could be ticketed separately. Or not if it doesn't make sense to you to separate them out.

> * Do we know what entering "other" does to the link we create?

Users should only be entering a link to a Globus endpoint here. I've asked Astrid to update the option text to "Other Globus endpoint" and the GDrive to "Stanford Google Drive" (non-Stanford will not work).

There is info above about what we need to compose the link based on the info about the endpoint. Karl Kornel should be able to assist with how to do this. We may need to restrict users to entering a link formatted a certain way to make sure it's to a Globus endpoint?

> * Can I split these into two (or three) tickets?

Sure. However you think makes the most sense.

**Comment by peetucket** [comment]
Get Astrid's input on:

- eliminate text box next to "other"
- reconsider use of radio buttons (allowing user to not do anything if not one of box/sherlock/gdrive)

Then update text descriptions of feature in ticket/docs to match and update designs.

**Comment by peetucket** [comment]
@astridu a few thoughts on this after we discussed:

- eliminate text box next to "other" (don't need it anymore)
- reconsider use of radio buttons (allowing user to not do anything if not one of oak/sherlock/gdrive)

Then update text descriptions of feature in ticket/docs to match and update designs.

**Comment by astridu** [comment]
@peetucket @amyehodge 
here are a couple of options. The second option doesn't require the user to select an option from the dropdown, but might just be more clear to have the radio buttons. The last option is more complicated in that if user selects 'Yes', then we have to present them with another set of radio buttons for them to select which one. Let me know which you prefer.

<img width="1073" alt="image" src="https://github.com/sul-dlss/happy-heron/assets/42216982/96839bfd-b972-4b02-a94f-31ef42aead35">
<img width="1073" alt="image" src="https://github.com/sul-dlss/happy-heron/assets/42216982/9a16ce3d-8bfa-4673-8385-76707053c408">
<img width="1073" alt="image" src="https://github.com/sul-dlss/happy-heron/assets/42216982/8dd72fbe-4ab6-4164-8196-4581606228b6">

**Comment by amyehodge** [comment]
@astridu @peetucket I feel like the question in the last option is the clearest, but the added complexity of the design is a drawback. Can we somehow combine these? Perhaps use the first design with the radio buttons, but the question would be "Are your files on Stanford Google Drive, Oak, or Sherlock?" And the choices could be:
- Yes, Stanford Google Drive
- Yes, Oak
- Yes, Sherlock
- No, they are somewhere else.

Are those choices too wordy?

**Comment by peetucket** [comment]
I think all of the 'yes' seems a a bit wordy to me.  I also like the compactness of the drop down menu vs radio buttons.

Another option:

"If your files are on Stanford Google Drive, Oak, or Sherlock, select it below."
-- drop down menu with those three options ---

If the user selects nothing, it's other.  So basically so action needed for other.  Even if it is one of those options and they do nothing, it doesn't close it off (they can still upload from those locations in the globus client, they just don't get a handy link shortcut).

**Comment by amyehodge** [comment]
Thanks @peetucket. That's better. I agree my suggestion was really wordy.

**Comment by peetucket** [comment]
So maybe if @astridu agrees, she can update the design, I can update descriptions, we can remove "design needed" and move to ready.

**Comment by astridu** [comment]
Yes, sorry for the delay, here it is.
<img width="1073" alt="image" src="https://github.com/sul-dlss/happy-heron/assets/42216982/1692a902-a492-41ab-a34b-4597acb68172">

**Comment by mjgiarlo** [comment]
@peetucket 💬 
> So maybe if @astridu agrees, she can update the design, I can update descriptions, we can remove "design needed" and move to ready.

It looks like we're good to go. I'll read through the comments and work on this today.

**Comment by mjgiarlo** [comment]
@astridu 💬 
> Yes, sorry for the delay, here it is.

Thank you!

**Comment by peetucket** [comment]
Thanks @astridu ... i updated the screenshot in the body of the ticket description to the new design.  If possible, would you mind updating https://projects.invisionapp.com/share/YQ132HXFX2GR#/screens/472346509 to avoid any potential future confusion (or at least mark that design as old)?

**Comment by astridu** [comment]
updated
https://projects.invisionapp.com/share/YQ132HXFX2GR#/screens/472346509

**Comment by amyehodge** [comment]
Google Drive: https://app.globus.org/file-manager?origin_id=e1c8858b-d5aa-4e36-b97e-95913047ec2b
Oak: https://app.globus.org/file-manager?origin_id=8b3a8b64-d4ab-4551-b37e-ca0092f769a7 
Sherlock: https://app.globus.org/file-manager?origin_id=6881ae2e-db26-11e5-9772-22000b9da45e
